src/abb_system_position_only.cpp

```diff
@@ -14,264 +14,259 @@
+      return CallbackReturn::ERROR;
+    }
+
+    if (joint.state_interfaces[0].name != hardware_interface::HW_IF_POSITION) {
+      RCLCPP_FATAL(
+        rclcpp::get_logger("ABBSystemPositionOnlyHardware"),
+        "expecting only POSITION state interface");
+      return CallbackReturn::ERROR;
+    }
+  }
+
+  // Configure EGM
+  RCLCPP_INFO(rclcpp::get_logger("ABBSystemPositionOnlyHardware"), "Configuring EGM interface...");
+
+  egm_interface_ =
+    std::make_unique<abb::egm::EGMControllerInterface>(io_service_, robotstudio_port_);
+  if (!egm_interface_) {
+    RCLCPP_FATAL(
+      rclcpp::get_logger("ABBSystemPositionOnlyHardware"),
+      "Could not create EGMControllerInterface");
+    return CallbackReturn::ERROR;
+  }
+
+  if (!egm_interface_->isInitialized()) {
+    RCLCPP_FATAL(
+      rclcpp::get_logger("ABBSystemPositionOnlyHardware"),
+      "EGM UDP Server interface failed to initialize (e.g. due to port already bound)");
+    return CallbackReturn::ERROR;
+  }
+
+  // done
+  return CallbackReturn::SUCCESS;
 }
 
-
-CallbackReturn ABBSystemPositionOnlyHardware::on_configure(const rclcpp_lifecycle::State &  /* previous_state */)
+CallbackReturn ABBSystemPositionOnlyHardware::on_configure(
+  const rclcpp_lifecycle::State & /* previous_state */)
 {
-	RCLCPP_INFO(rclcpp::get_logger("ABBSystemPositionOnlyHardware"),
-	"on_configure()");
+  RCLCPP_INFO(rclcpp::get_logger("ABBSystemPositionOnlyHardware"), "on_configure()");
 
-	//just in case - not 100% sure this is the right thing to do . . .
-	for (size_t i = 0; i < hw_states_.size(); ++i){
-		hw_states_[i] = std::numeric_limits<double>::quiet_NaN();
-		hw_commands_[i] = std::numeric_limits<double>::quiet_NaN();
-	}
-
-	return CallbackReturn::SUCCESS;
+  // just in case - not 100% sure this is the right thing to do . . .
+  for (size_t i = 0; i < hw_states_.size(); ++i) {
+    hw_states_[i] = std::numeric_limits<double>::quiet_NaN();
+    hw_commands_[i] = std::numeric_limits<double>::quiet_NaN();
+  }
 
+  return CallbackReturn::SUCCESS;
 }
 
-std::vector<hardware_interface::StateInterface> ABBSystemPositionOnlyHardware::export_state_interfaces()
+std::vector<hardware_interface::StateInterface>
+ABBSystemPositionOnlyHardware::export_state_interfaces()
 {
-	RCLCPP_INFO(rclcpp::get_logger("ABBSystemPositionOnlyHardware"),
-	"export_state_interfaces()");
-
-	std::vector<hardware_interface::StateInterface> state_interfaces;
-	for(size_t i=0; i<info_.joints.size(); i++){
-		state_interfaces.emplace_back(
-				hardware_interface::StateInterface(
-					info_.joints[i].name,
-					hardware_interface::HW_IF_POSITION,
-					&hw_states_[i]));
-	}
-	return state_interfaces;
+  RCLCPP_INFO(rclcpp::get_logger("ABBSystemPositionOnlyHardware"), "export_state_interfaces()");
+
+  std::vector<hardware_interface::StateInterface> state_interfaces;
+  for (size_t i = 0; i < info_.joints.size(); i++) {
```

Comment:
```suggestion
  for (size_t i = 0; i < info_.joints.size(); ++i) {
```

Very nitpicky, I know...